<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>add_book_copy</class>
 <widget class="QMainWindow" name="add_book_copy">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>469</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
QWidget { background: #f3f6ff; color: #0f172a; }


QLineEdit, QComboBox, QSpinBox, QDateEdit {
  background: #ffffff;
  color: #0f172a;
  border: 1px solid #e0e2eb;
  border-radius: 8px;
  padding: 6px 8px;
}
QLineEdit:focus, QComboBox:focus, QSpinBox:focus, QDateEdit:focus {
  border-color: #617fb9;            
  outline: none;
}


QPushButton {
  background: #ffffff;
  color: #4568ad;
  border: 2px solid #4568ad;
}
QPushButton:hover   { background: #e9ecf5; border-color: #3e5e9c; }
QPushButton:pressed { background: #d6d8e0; border-color: #37538a; }
QPushButton:disabled{
  background: #f1f3f9; color: #9aa0a6; border-color: #e0e2eb;
}

QPushButton#primaryBtn { background:#4568ad; color:#fff; border-color:#4568ad; }
QPushButton#primaryBtn:hover   { background:#3e5e9c; }
QPushButton#primaryBtn:pressed { background:#37538a; }

QTableView {
  background: #ffffff;
  alternate-background-color: #e9ecf5;
  gridline-color: #e0e2eb;
  selection-background-color: #4568ad;
  selection-color: #ffffff;
}


QToolTip {
  color: #ffffff;
  background: #1f2937;
  border: 1px solid #0b1220;
  border-radius: 8px;
  padding: 6px;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QLabel" name="label">
      <property name="font">
       <font>
        <pointsize>15</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Thêm bản sao sách</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="verticalWidget" native="true">
      <property name="styleSheet">
       <string notr="true">QWidget{background-color: #ffffff;
border: 2px solid #E0E0E0;
border-radius: 12px;
}
QLabel{
	border: 0px solid #E0E0E0;
}
QTableWidget{
	border-radius: 0px;
	border: 0px;
}</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="leftMargin">
        <number>15</number>
       </property>
       <property name="topMargin">
        <number>15</number>
       </property>
       <property name="rightMargin">
        <number>15</number>
       </property>
       <property name="bottomMargin">
        <number>15</number>
       </property>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Tên đầu sách</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="ten_dau_sach">
         <property name="styleSheet">
          <string notr="true">QComboBox {
    background-color: #f3f4f6;
    border: 1px solid #f3f4f6;
    border-radius: 8px;
    padding: 6px 12px;
    color: #333;
}

QComboBox:hover {
    background-color: #e4e6eb;
    border-color: #e4e6eb;
}

QComboBox:on {
    background-color: #e4e6eb;
    border-bottom-left-radius: 0px;
    border-bottom-right-radius: 0px;
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 30px;
    border-left-width: 0px;
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
}

QComboBox::down-arrow {
    image: url(:/icons/icons_/down-chevron.png);
    width: 12px;
    height: 12px;
}

QComboBox QAbstractItemView {
    background-color: white;
    border: 1px solid #dcdcdc;
    border-radius: 8px;
    border-top-left-radius: 0px;
    border-top-right-radius: 0px;
    color: #333;
    selection-background-color: #f3f4f6;
    selection-color: #333;
    outline: 0;
}

QComboBox QAbstractItemView::item {
    padding: 8px 12px;
    min-height: 25px;
}

QComboBox QAbstractItemView::item:hover {
    background-color: #f0f5ff;
    color: rgb(60, 91, 149);
}

QComboBox QAbstractItemView::item:selected {
    background-color: #e4e6eb;
    color: #333;
}</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Số lượng bản sao</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="so_ban_sao">
         <property name="styleSheet">
          <string notr="true">/* --- PHẦN CHÍNH (Đồng bộ với QLineEdit) --- */
QSpinBox, QDoubleSpinBox {
    background-color: #e8ecef; 
    border-radius: 8px; 
    border: 1px solid #E0E4E8; 
    padding: 8px 10px; 
    /* Quan trọng: chừa chỗ bên phải để chứa 2 nút bấm, không chữ sẽ bị che */
    padding-right: 30px; 
    color: #1E1E1E; 
    font-size: 12px;
}

QSpinBox:hover, QDoubleSpinBox:hover {
    background-color: #E5E9ED;
}

QSpinBox:focus, QDoubleSpinBox:focus {
    background-color: #FFFFFF; 
    border: 1px solid #3498db; 
}

/* --- PHẦN NÚT BẤM (Up/Down) --- */

/* Nút LÊN */
QSpinBox::up-button, QDoubleSpinBox::up-button {
    subcontrol-origin: border;
    subcontrol-position: top right; /* Nằm góc trên phải */
    width: 25px; /* Chiều rộng vùng bấm */
    
    /* Đường kẻ ngăn cách giữa số và nút bấm */
    border-left: 1px solid #E0E4E8;
    border-bottom: 1px solid #E0E4E8; /* Ngăn cách nút lên/xuống */
    
    /* Bo góc chỉ ở bên phải trên cùng để khớp với khung chính */
    border-top-right-radius: 8px; 
    background-color: transparent; /* Trong suốt để ăn theo màu nền chính */
}

/* Nút XUỐNG */
QSpinBox::down-button, QDoubleSpinBox::down-button {
    subcontrol-origin: border;
    subcontrol-position: bottom right; /* Nằm góc dưới phải */
    width: 25px;
    
    border-left: 1px solid #E0E4E8;
    border-top: 0px; /* Không cần viền trên vì dính với nút up rồi */
    
    /* Bo góc dưới phải */
    border-bottom-right-radius: 8px;
    background-color: transparent;
}

/* Hiệu ứng khi rê chuột vào nút bấm (làm đậm xíu cho biết đang bấm) */
QSpinBox::up-button:hover, QSpinBox::down-button:hover,
QDoubleSpinBox::up-button:hover, QDoubleSpinBox::down-button:hover {
    background-color: #d1d5da; /* Đậm hơn chút so với nền hover của khung */
}

/* Mũi tên (Chỉnh màu cho hợp tông text #1E1E1E) */
QSpinBox::up-arrow, QDoubleSpinBox::up-arrow,
QSpinBox::down-arrow, QDoubleSpinBox::down-arrow {
    width: 8px;
    height: 8px;
    /* Mẹo: Dùng màu border để tạo mũi tên nếu không có ảnh, 
       nhưng mặc định Qt sẽ tự render màu đen xám, khá hợp rồi */
}</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Danh sách bản sao</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTableWidget" name="danh_sach_ban_sao">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>200</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QScrollArea {
    border: none;
    background-color: transparent;
}

QScrollBar:vertical {
    border: none;
    background: transparent;
    width: 8px;
    margin: 0px 0px 0px 0px;
}

QScrollBar::handle:vertical {
    background-color: #d1d5db;
    min-height: 30px;
    border-radius: 4px;
}

QScrollBar::handle:vertical:hover {
    background-color: #9ca3af;
}

QScrollBar::sub-line:vertical {
    border: none;
    background: none;
    height: 0px;
}

QScrollBar::add-line:vertical {
    border: none;
    background: none;
    height: 0px;
}

QScrollBar::up-arrow:vertical, QScrollBar::down-arrow:vertical {
    background: none;
}

QScrollBar::add-page:vertical, QScrollBar::sub-page:vertical {
    background: none;
}
QTableWidget {
    background: #f7fbff;              /* nền sáng xanh nhạt */
    gridline-color: #d0e3ff;          /* màu kẻ ô */
    selection-background-color: #d6e9ff; /* nền khi chọn */
    selection-color: #0f172a;         /* màu chữ khi chọn */
    border: 0px;
}

QHeaderView::section {
    background: #e5f0ff;              /* header xanh nhạt */
    color: #0f172a;
    padding: 6px 10px;
    border: 0px;
    border-right: 1px solid #d0e3ff;
    border-bottom: 1px solid #d0e3ff;
    font-weight: 600;
}

/* kẻ ngang từng dòng */
QTableWidget::item {
    border-bottom: 1px solid #eef2ff;
}
QTableWidget::item:selected {
    background: #d6e9ff;
}
QWidget{
	border: 0px;
	border-radius: 0px;
}</string>
         </property>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderMinimumSectionSize">
          <number>30</number>
         </attribute>
         <column>
          <property name="text">
           <string>Mã bản sao</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Tên đầu sách</string>
          </property>
         </column>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="them_button">
           <property name="styleSheet">
            <string notr="true">QPushButton {
    background-color: #3498db; /* Xanh dương tươi (Bright Blue) */
    color: white;
    border: none;
    border-radius: 8px;
    padding: 10px 20px;
    font-weight: bold;
    font-size: 14px;
}

QPushButton:hover {
    background-color: #5dade2; /* Sáng hơn khi di chuột */
}

QPushButton:pressed {
    background-color: #2980b9; /* Tối hơn chút khi ấn */
    padding-top: 11px;
}
</string>
           </property>
           <property name="text">
            <string>Thêm</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
